src/main/kotlin/org/rust/ide/hints/codeVision/RsReferenceCodeVisionProvider.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.hints.codeVision
+
+import com.intellij.codeInsight.codeVision.CodeVisionRelativeOrdering
+import com.intellij.codeInsight.hints.codeVision.ReferencesCodeVisionProvider
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.RsBundle
+import org.rust.lang.core.psi.RsFile
+import org.rust.lang.core.psi.ext.RsElement
+import org.rust.lang.core.psi.ext.RsQualifiedNamedElement
+import org.rust.lang.core.psi.ext.searchReferences
+
+class RsReferenceCodeVisionProvider : ReferencesCodeVisionProvider() {
+    override fun acceptsFile(file: PsiFile): Boolean = file is RsFile
+
+    override fun acceptsElement(element: PsiElement): Boolean = element is RsQualifiedNamedElement
```

Comment:
I think we should ignore certain `RsQualifiedNamedElement`s here, particularly `RsFile` and `RsModDeclItem`.
Let's enumerate all the elements that we want to process (the same way we did in `RsVcsCodeVisionContext`) and cover all of them with some tests.